crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1681,6 +1723,255 @@ impl<'src> Parser<'src> {
         }
     }
 
+    /// Parses a t-string.
+    ///
+    /// This does not handle implicitly concatenated strings.
+    ///
+    /// # Panics
+    ///
+    /// If the parser isn't positioned at a `TStringStart` token.
+    ///
+    /// See: <https://docs.python.org/3/reference/grammar.html> (Search "fstring:")
+    /// See: <https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals>
+    fn parse_tstring(&mut self) -> ast::TString {
```

Comment:
What's the reason that we need to buplicate the entire logic here? Can't we share some logic with parsing f-strings, given that they're, as far as I understand, identical?